airflow/sensors/base.py

```diff
@@ -186,11 +208,44 @@ def __init__(
         self.mode = mode
         self.exponential_backoff = exponential_backoff
         self.max_wait = self._coerce_max_wait(max_wait)
-        if soft_fail is True and never_fail is True:
-            raise ValueError("soft_fail and never_fail are mutually exclusive, you can not provide both.")
+        if skip_policy != NOTSET:
+            if sum([soft_fail, silent_fail]) > 0:
+                raise ValueError(
+                    "skip_policy and deprecated soft_fail and silent_fail parameters are mutually exclusive."
+                )
+
+            if skip_policy == SkipPolicy.SKIP_ON_SOFT_ERROR:
+                self.soft_fail = True
```

Comment:
If we're already using `skip_policy` to check. Do we still need this assignment here?

Reply:
some sensors still use self.soft_fail